packages/@n8n/decorators/src/metadata-state.ts

```diff
@@ -0,0 +1,36 @@
+import { Service } from '@n8n/di';
+
+import type { Controller, ControllerMetadata, HandlerName, RouteMetadata } from './types';
+
+@Service()
+export class MetadataState {
```

Comment:
Sorry, I should have named this better. Can we please rename this to `ControllerRegistryMetadata` or `RouteRegistryMetadata`.